belgif-rest-problem-it/belgif-rest-problem-quarkus-it/src/main/java/io/github/belgif/rest/problem/quarkus/it/FrontendImpl.java

```diff
@@ -0,0 +1,232 @@
+package io.github.belgif.rest.problem.quarkus.it;
+
+import java.net.URI;
+import java.util.concurrent.ExecutionException;
+
+import jakarta.annotation.PostConstruct;
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.inject.Inject;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Positive;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.QueryParam;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.Response;
+
+import org.eclipse.microprofile.rest.client.RestClientBuilder;
+
+import com.acme.custom.CustomProblem;
+
+import io.github.belgif.rest.problem.BadRequestProblem;
+import io.github.belgif.rest.problem.DefaultProblem;
+import io.github.belgif.rest.problem.ServiceUnavailableProblem;
+import io.github.belgif.rest.problem.api.Problem;
+import io.github.belgif.rest.problem.model.ChildModel;
+import io.github.belgif.rest.problem.model.Model;
+import io.github.belgif.rest.problem.model.NestedModel;
+import io.vertx.core.http.HttpServerRequest;
+
+@RequestScoped
+@Path("/frontend")
+public class FrontendImpl implements Frontend {
+
+    @Context
+    private HttpServerRequest serverRequest;
+
+    private URI baseUri;
+
+    private Backend microprofileClient;
+
+    private jakarta.ws.rs.client.Client jaxRsClient;
+
+    @PostConstruct
+    public void initialize() {
+        this.baseUri = URI.create("http://" + serverRequest.localAddress().host() + ":"
+                + serverRequest.localAddress().port() + "/quarkus");
+        this.microprofileClient = RestClientBuilder.newBuilder()
```

Comment:
Quarkus [REST client guide](https://quarkus.io/guides/rest-client) describes also these 2 ways of creating clients:
* `@RegisterRestClient` annotated interface injected in resource class with `@RestClient`
* Using the `QuarkusRestClientBuilder` 

While underlying microprofile is used verifying that constructed through these different method behavior is still the same adds a bit of certainty (properly configured for all methods described in the Quarkus guides).